x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/rules_list/components/notify_badge/notify_badge.tsx

```diff
@@ -378,5 +413,78 @@ export const RulesListNotifyBadge: React.FunctionComponent<RulesListNotifyBadgeP
       </EuiFlexGroup>
     );
   }
+
   return popover;
 };
+
+interface InvalidSnoozeButtonProps {
+  isLoading: boolean;
+  isDisabled: boolean;
+  ref?: React.RefObject<HTMLButtonElement>;
+  onClick: () => void;
+}
+
+const InvalidSnoozeButton: React.FC<InvalidSnoozeButtonProps> = memo(
+  ({ isLoading, isDisabled, onClick, ref }) => {
+    return (
+      <EuiButton
+        size="s"
+        isLoading={isLoading}
+        disabled={isLoading || isDisabled}
+        data-test-subj="rulesListNotifyBadge-invalidSnooze"
+        aria-label={INVALID_SNOOZE_TOOLTIP_TITLE}
+        minWidth={85}
+        iconType="warning"
+        color="danger"
+        onClick={onClick}
+        buttonRef={ref}
+      >
+        <EuiText size="xs">{INVALID_SNOOZE}</EuiText>
+      </EuiButton>
+    );
+  }
+);
+
+const isSnoozeScheduleValid = (snoozeSchedule: RuleSnoozeSettings['snoozeSchedule']) => {
+  if (snoozeSchedule == null || snoozeSchedule.length === 0) {
+    return true;
+  }
+
+  return snoozeSchedule.every(isSnoozeValid);
+};
+
+const isSnoozeValid = (snooze: NonNullable<RuleSnoozeSettings['snoozeSchedule']>[number]) => {
+  const { rRule } = snooze;
+
+  const rRuleOptions = {
+    dtstart: rRule.dtstart,
+    tzid: rRule.tzid,
+    freq: rRule.freq,
+    interval: rRule.interval,
+    until: rRule.until,
+    count: rRule.count,
+    byweekday: rRule.byweekday,
+    bymonthday: rRule.bymonthday,
+    bymonth: rRule.bymonth,
+  };
+
+  return isValidateRRule(rRuleOptions);
+};
+
+const getSnoozeScheduleIds = (snooze: NonNullable<RuleSnoozeSettings['snoozeSchedule']> = []) => {
+  return snooze.map(({ id }) => id).filter(Boolean) as string[];
+};
+
+const isValidateRRule = (rRule: RRuleParams): boolean => {
+  const validWeekDays = new Set(['MO', 'TU', 'WE', 'TH', 'FR', 'SA', 'SU']);
+
+  if (moment.tz.zone(rRule.tzid) == null) {
+    return false;
+  }
+
+  if (rRule.byweekday && !rRule.byweekday.every((value) => validWeekDays.has(value as string))) {
```

Comment:
I think `rRule` package also verifies if `byweekday ` is from valid weekdays or not, do we still need this check here?